jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1052,14 +1065,22 @@ else if (_endOfContent == EndOfContent.CHUNKED_CONTENT)
                     if (_fieldCache == null)
                     {
                         _fieldCache = (getHeaderCacheSize() > 0 && (_version != null && _version == HttpVersion.HTTP_1_1))
-                            ? new ArrayTernaryTrie<>(getHeaderCacheSize())
+                            ? new Index.Builder<HttpField>()
+                            .caseSensitive(false)
+                            .mutable()
+                            .maxCapacity(getHeaderCacheSize())
+                            .build()
```

Comment:
See earlier comment about the expense of this. We need to make this as cheap as possible:
```suggestion
                            ? new Index.MutableBuilder<HttpField>()
                            .caseSensitive(false)
                            .maxCapacity(getHeaderCacheSize())
                            .build()
```

Reply:
See above.